lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2968,17 +3042,20 @@ decodeSharedTransaction ctx (ApiT wid) (ApiSerialisedTransaction (ApiT sealed) _
         -- TODO minting/burning multisig
         , mint = emptyApiAssetMntBurn
         , burn = emptyApiAssetMntBurn
-        -- TODO delegation/withdrawals multisig
-        , certificates = []
-        , depositsTaken = []
-        , depositsReturned = []
+        , certificates = certs
+        , depositsTaken =
+            (Quantity . fromIntegral . unCoin . W.stakeKeyDeposit $ pp)
+                <$ filter ourRewardAccountRegistration certs
+        , depositsReturned =
+            (Quantity . fromIntegral . unCoin . W.stakeKeyDeposit $ pp)
```

Comment:
Suggestion: use `Coin.toQuantity` (this is a statically-checked total function).

```suggestion
            (Coin.toQuantity . W.stakeKeyDeposit $ pp)
```